src/Validateable.ts

```diff
@@ -5,25 +5,37 @@ export default class Validateable {
   }
 
   number(): number {
-    if (typeof this.value !== "number") {
-      throw new Error(`${this.value} is not a number`);
+    if (typeof this.value === "number") {
+      if (isNaN(this.value)) throw new Error(`${this.value} is NaN`);
+      return this.value;
     }
-    if (isNaN(this.value)) throw new Error(`${this.value} is NaN`);
 
-    return this.value;
+    if (typeof this.value === "string" && this.value.length > 0) {
+      console.log("here");
+      if (this.value.match(/^-{0,1}\d+$/)) {
```

Comment:
use this instead `/^[+-]?\d+\.?\d+$/` to match decimal numbers and positive sign.